compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -957,6 +963,9 @@ public void visit(BLangInvocation iExpr) {
         }
 
         BType varRefType = iExpr.expr.type;
+
+        checkSafeNavigation(iExpr, varRefType);
+
         varRefType = getSafeType(varRefType, iExpr);
```

Comment:
We should get the safe type only if checkSafeNavigation() check passes...
Otherwise `varRefType = symTable.semanticError`

Same for all places

Reply:
Changed with https://github.com/ballerina-platform/ballerina-lang/pull/12507/commits/0e634983cff6a5fdd707a28026920b7f4de7d3e4.